src/app/dataset-flow/dataset-flow-details/tabs/flow-details-history-tab/flow-details-history-tab.component.html

```diff
@@ -0,0 +1,47 @@
+<div class="container">
+    <div class="row fw-500">
+        <div class="col-md-6">Event</div>
+        <div class="col-md-3">Time</div>
+        <div class="col-md-3">Duration</div>
+    </div>
+    <mat-divider class="my-2"></mat-divider>
+    <ul>
+        <li *ngFor="let flowEvent of flowHistory; let lastIndex = last; let i = index">
+            <div class="mt-4">
+                <div class="row">
+                    <div class="col-md-6 d-flex position-relative">
+                        <div class="history-item">
+                            <img
+                                *ngIf="lastIndex && flowDetails.status !== FlowStatus.Finished"
+                                src="assets/images/hourglass.gif"
+                                alt="hourglass"
+                                style="width: 15px"
+                                class="hourglass-img text-muted"
+                            />
+                            <mat-icon
+                                class="icon-status d-block"
+                                [ngClass]="flowEventIconOptions(flowEvent, flowDetails).class"
+                            >
+                                {{ flowEventIconOptions(flowEvent, flowDetails).icon }}
+                            </mat-icon>
```

Comment:
Minor - do we have to compute the same info twice to get different attributes?

Reply:
Yes, is there another option?